bokeh/application/handlers/notebook.py

```diff
@@ -32,17 +32,53 @@ def __init__(self, *args, **kwargs):
             filename (str) : a path to a Jupyter notebook (".ipynb") file
 
         '''
-        nbformat = import_required('nbformat', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
-        nbconvert = import_required('nbconvert', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
-
         if 'filename' not in kwargs:
             raise ValueError('Must pass a filename to NotebookHandler')
         filename = kwargs['filename']
 
+        preprocessors = self._preprocessors()
+        try:
+            source, meta = self.export_python(filename, preprocessors)
+        except:
+            # If any preprocessors resulted in an exception, convert without them
+            source, meta = self.export_python(filename, preprocessors=[])
+
+        kwargs['source'] = source
+        super(NotebookHandler, self).__init__(*args, **kwargs)
+
+
+    def export_python(self, filename, preprocessors):
+        '''
+        Given an .ipynb notebook, convert to plain Python syntax using
+        the supplied list of nbconvert preprocessors.
+        '''
+        nbformat = import_required('nbformat', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
+        nbconvert = import_required('nbconvert', 'The Bokeh notebook application handler requires Jupyter Notebook to be installed.')
+
         with open(filename) as f:
             nb = nbformat.read(f, nbformat.NO_CONVERT)
             exporter = nbconvert.PythonExporter()
-            source, meta = exporter.from_notebook_node(nb)
-            kwargs['source'] = source
+            for preprocessor in preprocessors:
+                exporter.register_preprocessor(preprocessor)
+            return exporter.from_notebook_node(nb)
```

Comment:
Is this function really necessary? Esp if the try/except above can be removed.  It doesn't refer to anything though `self` why is it a method not a free function? Alternatively it could use `self.preprocessors` directly and not take it as a parameter.